iceoryx2/src/node/mod.rs

```diff
@@ -364,6 +364,28 @@ impl<Service: service::Service> NodeState<Service> {
     }
 }
 
+#[allow(missing_docs)]
+impl<Service: service::Service> NodeState<Service> {
+    pub fn to_serializable(&self) -> SerializableNodeState {
+        match self {
+            NodeState::Alive(view) => SerializableNodeState::Alive(view.to_serializable()),
+            NodeState::Dead(view) => SerializableNodeState::Dead(view.to_serializable()),
+            NodeState::Inaccessible(id) => SerializableNodeState::Inaccessible(*id),
+            NodeState::Undefined(id) => SerializableNodeState::Undefined(*id),
+        }
+    }
+}
+
+#[allow(missing_docs)]
+#[derive(serde::Serialize)]
```

Comment:
Could you please move this into a new file `serializable_node_state.rs`. The reason is, that I suspect that we cannot use serialization in an ASIL D target so it would be nice to separate it a bit right from the beginning.